drivers/gpio/gpio_sn74hc595.c

```diff
@@ -151,50 +156,74 @@ static DEVICE_API(gpio, gpio_sn74hc595_drv_api_funcs) = {
  */
 static int gpio_sn74hc595_init(const struct device *dev)
 {
-	const struct gpio_sn74hc595_config *config = dev->config;
+	int ret;
 	struct gpio_sn74hc595_drv_data *drv_data = dev->data;
+	const struct gpio_sn74hc595_config *config = dev->config;
 
 	if (!spi_is_ready_dt(&config->bus)) {
 		LOG_ERR("SPI bus %s not ready", config->bus.bus->name);
 		return -ENODEV;
 	}
 
-	if (!gpio_is_ready_dt(&config->reset_gpio)) {
-		LOG_ERR("GPIO port %s not ready", config->reset_gpio.port->name);
-		return -ENODEV;
+	if (config->enable_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->enable_gpio)) {
+			LOG_ERR("GPIO port %s not ready", config->enable_gpio.port->name);
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(&config->enable_gpio, GPIO_OUTPUT_INACTIVE) < 0) {
+			LOG_ERR("Unable to configure ENABLE GPIO pin %u", config->enable_gpio.pin);
+			return -EINVAL;
+		}
 	}
 
-	if (gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_ACTIVE) < 0) {
-		LOG_ERR("Unable to configure RST GPIO pin %u", config->reset_gpio.pin);
-		return -EINVAL;
+	if (config->reset_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->reset_gpio)) {
+			LOG_ERR("GPIO port %s not ready", config->reset_gpio.port->name);
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_ACTIVE) < 0) {
+			LOG_ERR("Unable to configure RST GPIO pin %u", config->reset_gpio.pin);
+			return -EINVAL;
+		}
+		gpio_pin_set_dt(&config->reset_gpio, 0);
```

Comment:
Do we need a delay between the two to honor a minimum reset assertion time?

Reply:
There's also the possibility of a slight optimization that can be made during initializtion when `reset_value` is equal to zero: instead of using `sn74hc595_write` to shift in all of those zeros, we can just toggle `RCLK` from low to high after setting `SRCLR` low. @henrikbrixandersen Do you think that's a worthwhile optimization?